0-module-and-module-source.emu

```diff
@@ -548,8 +550,11 @@ location: https://tc39.es/proposal-compartments/
                 1. Else, if Type(_importMeta_) is not Object, then
                   1. Throw a *TypeError* exception.
                 1. Perform ? RequireInternalSlot(_moduleSource_, [[ModuleSource]]).
+                1. Let _evalRealm_ be the current Realm Record.
+                1. NOTE: Module Instances created from this constructor must use a valid source.
+                1. Perform ? HostEnsureCanCompileStrings(_evalRealm_).
```

Comment:
This should be called in ModuleSource constructor, because create a new module instance from a compiled source is safe, but convert a string to a ModuleSource is unsafe.

Reply:
This is a good point... but the problem that I see here is that csp can completely block the parsing of a source... let's chat more about it. Ideally, we can decorate the source record to conditionally call this host hook. cc @nicolo-ribaudo

Reply:
> csp can completely block the parsing of a source

Yes, that's how CSP works.

Reply:
Yes, I understand that's how it works today, but if we are going to extend the ModuleSource API to expose the metadata of the parsed source, you will not be able to parse it in the first place even though you don't plan to evaluate it. E.g.: a bundler.